feat(provider): add hetznercloud (#1046)
qdm12/ddns-updater: docs/hetznernetworking.md

# Hetzner Networking

## Configuration

### Example

```json
{
  "settings": [
    {
      "provider": "hetznernetworking",
      "zone_identifier": "example.com",

>       "domain": "example.com",

Reviewer: can we extract zone_identifier from domain instead? i.e. `x.y.com` -> `y.com` - it looks like `extractRRName` does it?

      "ttl": 600,
      "token": "yourtoken",
      "ip_version": "ipv4",
      "ipv6_suffix": ""
    }
  ]
}
```

### Compulsory parameters

- `"zone_identifier"` is the DNS zone name (e.g., `example.com`), not a zone ID
- `"domain"` is the domain to update. It can be `example.com` (root domain), `sub.example.com` (subdomain of `example.com`) or `*.example.com` for the wildcard.
- `"ttl"` optional integer value corresponding to a number of seconds

> - One of the following ([how to find API keys](https://docs.hetzner.cloud/api/getting-started/generating-api-token)):
>   - API Token `"token"`, configured with DNS write permissions for your DNS zone

Reviewer: ```suggestion
- `"token"` is your API token configured with DNS write permissions for your DNS zone, see [how to find API keys](https://docs.hetzner.cloud/api/getting-started/generating-api-token)
```
